crates/biome_js_analyze/src/lint/nursery/use_iterable_callback_return.rs

```diff
@@ -0,0 +1,378 @@
+use crate::ControlFlowGraph;
+use crate::services::control_flow::JsControlFlowGraph;
+use biome_analyze::{Rule, RuleDiagnostic, RuleSource, context::RuleContext, declare_lint_rule};
+use biome_console::fmt::{Display, Formatter};
+use biome_console::markup;
+use biome_control_flow::builder::ROOT_BLOCK_ID;
+use biome_control_flow::{ExceptionHandlerKind, InstructionKind};
+use biome_js_syntax::{
+    AnyJsFunctionBody, JsArrowFunctionExpression, JsCallArgumentList, JsCallArguments,
+    JsCallExpression, JsFunctionExpression, JsReturnStatement, global_identifier,
+};
+use biome_rowan::{AstNode, AstSeparatedList, NodeOrToken, TextRange};
+use roaring::RoaringBitmap;
+use rustc_hash::FxHashMap;
+use std::io;
+use std::sync::LazyLock;
+
+declare_lint_rule! {
+    /// Enforce consistent return values in iterable callbacks.
+    ///
+    /// This rule ensures that callbacks passed to certain iterable methods either always return a value or never return a value, depending on the method's requirements.
```

Comment:
I thought that the rule was about enforcing the `return` to certain methods, but it seems that it also does the opposite for other methods.

As a bikeshed, what do you think of naming the rule `useConsistentIterableCallback`?

Reply:
I'm fine with changing the name. Is there an easy way to do it?

Reply:
@ematipico We typically use the `useConsistent` prefix for when users have a choice of some kind, usually between 2 stylistic things. I think the name is fine as is. Perhaps a better alternative would be `useCorrectIterableCallbackReturn`?

@mdevils Unfortunately, not really. What I usually do is create a new lint rule with the new name and transfer all the rule code over, rerunning codegen as necessary.

Reply:
It makes sense :)

Let's keep this name